Tools/MiniBrowser/SwiftUI/ContentView.swift

```diff
@@ -26,15 +26,56 @@
 import SwiftUI
 @_spi(Private) import WebKit
 
-struct ContentView: View {
+fileprivate struct UnifiedBar: View {
     @Environment(BrowserViewModel.self) private var viewModel
 
     @FocusState private var urlFieldIsFocused: Bool
 
+    var body: some View {
+        @Bindable var viewModel = viewModel
+
+        VStack(spacing: 0) {
+            TextField("URL", text: $viewModel.displayedURL)
+                .textContentType(.URL)
+                .onSubmit {
+                    urlFieldIsFocused = false
+                    viewModel.navigateToSubmittedURL()
+                }
+                .textFieldStyle(.roundedBorder)
+                .focused($urlFieldIsFocused)
+
+            ProgressView(value: viewModel.page.estimatedProgress, total: 1.0)
+                .padding(.horizontal, 2)
+                .padding(.top, -4)
+                .padding(.bottom, -8)
+        }
+        .frame(minWidth: 300)
+
+        if viewModel.page.isLoading {
+            Button {
+                viewModel.page.stopLoading()
+            } label: {
+                Image(systemName: "xmark")
+            }
+        } else {
+            Button {
+                viewModel.page.reload()
+            } label: {
+                Image(systemName: "arrow.clockwise")
+            }
+        }
+    }
+}
+
+struct ContentView: View {
+    @Environment(BrowserViewModel.self) private var viewModel
+
     @AppStorage("DefaultURL") private var defaultURL = "https://www.webkit.org"
 
     var body: some View {
         WebView_v0(viewModel.page)
+            .webViewAllowsBackForwardNavigationGestures()
```

Comment:
Kinda related to the above, but since this doesn’t do anything to change the default I’m not sure whether it’s useful to include. We could have a `Toggle` somewhere to properly exercise it maybe?

Reply:
it does do something; if it's excluded, the default value is false. but the default value of the parameter is true, so this sets the preference to true from false